### PR TITLE
Signed Foundatio.Redis

### PR DESCRIPTION
**Pull request by rerikson**
Creates a signed Foundatio.Redis project that references the StackExchange.Redis.StrongName package.

### CODE REVIEW

**Comment by niemyjski** [comment]
Would it be possible to just sign our original packages. There is no disadvantage to have the regular package sign and it is a lot less work to maintain two sets of project files.

**Comment by rerikson** [comment]
That's no problem at all for the main Foundatio repo (submitted a separate PR for that and I'll update that code to remove the separate projects), but we were hoping to have a separate signed project for Foundatio.Redis, so that Foundatio.Redis.Signed could reference StackExchange.Redis.Strongname, which is used by SignalR.Redis.  Yves discussed it [here](https://github.com/FoundatioFx/Foundatio/issues/31) .  What do you think?

**Comment by niemyjski** [comment]
It doesn't make sense to have both, a strong named package can reference a non strong named package (e.g., redis)

**Comment by rerikson** [comment]
So, if I updated the PR to have only one signed Foundatio.Redis package that references StackExchange.Redis.Strongname instead of StackExchange.Redis as it does today, would that be OK?

**Comment by niemyjski** [comment]
I don't think that would help any as foundatio would need to be signed. @ejsmith would it make sense to only sign .net core targets?

**Comment by rerikson** [comment]
Right, Foundatio would still need to be signed and we would need to wait for that to finish before updating Foundatio.Redis to be signed, but if we're going to strong-name Foundatio.Redis doesn't it make sense to reference the strong-named version of StackExchange.Redis?

**Comment by niemyjski** [comment]
Yeah, we'd need to update that reference to be strong named as well.

**Comment by niemyjski** [comment]
I'm thinking we may just want to investigate what it would take a to source import the async-ex packages we are using.

**Comment by rerikson** [comment]
I'd be happy to look into that.  I haven't done a source import before though.  Is it something like [this](https://gist.github.com/martinbuberl/b58fd967f271f32f51f50aee62e7332c)?

**Comment by niemyjski** [comment]
I'm leaning against externals. I think it makes much more sense to just have a powershell script to do it (or bring it in manually). So here is what we do with exceptionless.net and json.net (https://github.com/exceptionless/Exceptionless.Net/blob/master/src/Exceptionless/update-json.ps1)

There are some pros todo this
1. We can only bring in the types we need (this will take a tiny bit of work to see what we are using for primitives. I don't think we are using very much).
2. We place them in a folder and prefix the namespace with Foundatio. The huge plus on this is that it's internalized and we're not leaking these dependencies into other peoples apps (causing you todo type forwarding).
3. Easier to maintain than an external (you don't have to do submodule init every time you clone and you have greater control over namespaces and the like) and upgrade later (on your own time).

**Comment by rerikson** [comment]
Thanks for the advice on the source import.  I implemented it as you suggested and pushed it to the PR over in the Foundatio repo

**Comment by niemyjski** [comment]
@rerikson Left a comment on it, I think only thing missing is copying the license from the repo to the folder via the powershell script. We very much appreciate your help.

**Comment by niemyjski** [comment]
I'll try and get this merged in asap once the license is there.

**Comment by rerikson** [comment]
Happy to help and thanks for the opportunity to contribute!  I updated the PR to pull in the license via the powershell scripts, so I think we should be all set.

File: appveyor.yml [hunk 1]
```diff
@@ -16,9 +16,11 @@ before_build:
   - ps: .\build\Set-BuildVersion -Version $env:APPVEYOR_BUILD_VERSION -Suffix $env:VERSION_SUFFIX
   - nuget sources add -name Exceptionless -source https://www.myget.org/F/exceptionless/api/v3/index.json
   - appveyor-retry dotnet restore -v Minimal
+  - appveyor-retry dotnet restore .\src\Foundatio.Redis.Signed -v Minimal
```
Comment (niemyjski) [inline appveyor.yml]: I don't think these dot net restores were left over from when we had two project folders?

File: src/Foundatio.Redis.Signed/Foundatio.Redis.Signed.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <ItemGroup Label="Build">
+    <Compile Remove="..\Foundatio.Redis\obj\**\*" />
+    <Compile Remove="artifacts\**" />
+    <Compile Include="..\Foundatio.Redis\**\*.cs;" Exclude="..\Foundatio.Redis\**\AssemblyInfo.cs;..\Foundatio.Redis\obj\**\*;bin\**;obj\**;packages\**" />
+  </ItemGroup>
+  
+  <PropertyGroup Label="Signing">
+    <SignAssembly>true</SignAssembly>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
+    <AssemblyOriginatorKeyFile>..\Exceptionless.snk</AssemblyOriginatorKeyFile>
```
Comment (niemyjski) [inline src/Foundatio.Redis.Signed/Foundatio.Redis.Signed.csproj]: I'm honestly not sure how the build is passing right now as the snk file needs to be copied to the repo.

File: appveyor.yml (hunk 1 above)
Comment (rerikson) [inline appveyor.yml]: Yeah, good catch.  I removed them and updated the PR.

File: src/Foundatio.Redis.Signed/Foundatio.Redis.Signed.csproj (hunk 2 above)
Comment (rerikson) [inline src/Foundatio.Redis.Signed/Foundatio.Redis.Signed.csproj]: Maybe I'm missing something, but it looks to me like the Exceptionless.snk file is included in the PR.  Can you double check that it's not there?

**Comment by niemyjski** [comment]
Thanks for the PR

**Pull request merged**